src/Flex.php

```diff
@@ -198,7 +227,7 @@ public function executeAutoScripts(Event $event): void
         $this->io->write($this->postInstallOutput);
     }
 
-    private function filterPackageNames(PackageInterface $package, string $operation)
+    private function filterPackageNames(PackageInterface $package, string $operation): ?\Generator
```

Comment:
yes, using `yield` in a function makes it a generator function (no matter if the `yield` statement is actually reached) thus always return a `\Generator` instance.